qiskit/pulse/schedule.py

```diff
@@ -274,6 +274,30 @@ def parameters(self) -> Set:
         """Parameters which determine the schedule behavior."""
         return self._parameter_manager.parameters
 
+    def get_pulses(self, name: Optional[str] = None) -> List:
+        """
+        Get a list of pulses used in the schedule.
+
+        Args:
+            name: Name of the pulse to filter by. If provided, only pulses with a matching name
+                  will be returned. Defaults to None, which returns all pulses.
+
+        Returns:
+            List of pulse objects used in the schedule, filtered by name if specified.
+        """
+        if not name:
+            pulses = []
+            for instruction in self.instructions:
+                if hasattr(instruction[1], "pulse"):
```

Comment:
```suggestion
                matched = name is None or instruction[1].pulse.name == name
                if hasattr(instruction[1], "pulse") and matched:
```

Then you can drop another loop?